### PR TITLE
New run action apply-manifest

### PR DESCRIPTION
**Pull request by xiwenc**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Introduces a new action for kubernetes-master: apply-manifest takes a JSON string as input and applies it to k8s' like `kubectl apply -f FILE.json`. This action is useful to run additional cluster setup after the k8s cluster has been deployed. It's a general purpose action that can be used any time to change/manage the cluster via Juju.

**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:
This feature has been discussed with @tvansteenburgh in the past.

**Does this PR introduce a user-facing change?**:
```release-note
Add support for applying any kubernetes manifest via Juju action
```

Example usage:

```shell
ubuntu@juju-95e64e-0:~$ cat mysecret.json
{
  "apiVersion": "v1",
  "kind": "Secret",
  "metadata": {
    "name": "mysecret"
  },
  "type": "Opaque",
  "data": {
    "username": "YWRtaW4=",
    "password": "MWYyZDFlMmU2N2Rm"
  }
}
ubuntu@juju-95e64e-0:~$ juju run-action kubernetes-master/0 apply-manifest --wait --string-args json="$(cat mysecret.json)"
unit-kubernetes-master-0:
  id: 0220dccf-1727-4ae5-8db6-f4a8f542bd2d
  results:
    output: |
      secret/mysecret created
    summary: Manifest applied.
  status: completed
  timing:
    completed: 2019-04-08 09:22:08 +0000 UTC
    enqueued: 2019-04-08 09:22:05 +0000 UTC
    started: 2019-04-08 09:22:07 +0000 UTC
  unit: kubernetes-master/0
ubuntu@juju-95e64e-0:~$ juju run-action kubernetes-master/0 apply-manifest --wait --string-args json="$(cat mysecret.json)"
unit-kubernetes-master-0:
  id: 0266600c-6c8e-4d6b-8ce4-1924740b62db
  results:
    output: |
      secret/mysecret unchanged
    summary: Manifest applied.
  status: completed
  timing:
    completed: 2019-04-08 09:22:15 +0000 UTC
    enqueued: 2019-04-08 09:22:13 +0000 UTC
    started: 2019-04-08 09:22:15 +0000 UTC
  unit: kubernetes-master/0
```

### CODE REVIEW

File: actions.yaml [hunk 1]
```diff
@@ -50,3 +50,12 @@ upgrade:
     description: Upgrade the kubernetes snaps
 get-kubeconfig:
   description: Retrieve Kubernetes cluster config, including credentials
+apply-manifest:
+  description: Apply JSON formatted Kubernetes manifest to cluster
+  params:
+    json:
+      type": string
```
Comment (hyperbolic2346) [inline actions.yaml]: typo "

File: actions/kubectl-actions.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        action_fail("Failed to retrieve kubeconfig: {}".format(str(e)))
+
+
+def apply_manifest():
+    """
+    Applies a user defined manifest with kubectl
+    """
+    _, apply_path = tempfile.mkstemp(suffix=".json")
+    try:
+        manifest = json.loads(action_get("json"))
```
Comment (hyperbolic2346) [inline actions/kubectl-actions.py]: Is the reasoning behind the json.loads -> json.dump to try and catch formatting issues before kubectl apply in the hopes that json.loads would give better error messages?

File: actions.yaml (hunk 1 above)
Comment (xiwenc) [reply actions.yaml]: Thanks for spotting this

File: actions/kubectl-actions.py (hunk 2 above)
Comment (xiwenc) [reply actions/kubectl-actions.py]: Yeah that is the indeed the reasoning.

**Comment by Cynerva** [comment]
LGTM

**Pull request merged**